lib/Yield/PTML.cpp

```diff
@@ -132,28 +163,36 @@ static std::set<std::string> targets(const yield::BasicBlock &BasicBlock,
     .OptionalCallAddress = MetaAddress::invalid()
   };
 
+  std::string NextBlock = targetPath(BasicBlock.nextBlock(), Function, Binary);
+
   std::set<std::string> Result;
   for (const auto &Edge : BasicBlock.Successors()) {
     auto TargetPair = efa::parseSuccessor(*Edge,
                                           BasicBlock.nextBlock(),
                                           Binary);
-    if (TargetPair.NextInstructionAddress.isValid()) {
+    if (TargetPair.NextInstructionAddress.isValid()
+        && TargetPair.NextInstructionAddress != BasicBlock.nextBlock()) {
       std::string Path = targetPath(TargetPair.NextInstructionAddress,
                                     Function,
                                     Binary);
-      if (!Path.empty()) {
+      if (!Path.empty())
         Result.insert(Path);
-      }
     }
-    if (TargetPair.OptionalCallAddress.isValid()) {
+    if (TargetPair.OptionalCallAddress.isValid()
+        && TargetPair.NextInstructionAddress != BasicBlock.nextBlock()) {
       BasicBlockID ID = BasicBlockID(TargetPair.OptionalCallAddress);
       std::string Path = targetPath(ID, Function, Binary);
-      if (!Path.empty()) {
+      if (!Path.empty())
         Result.insert(Path);
-      }
     }
   }
 
+  // Only keep next instruction target if there are multiple other targets,
+  // i.e. it's a switch. This lets us avoid it in cases where it's just a call
+  // or a conditional jump.
+  if (Result.size() != 1)
+    Result.emplace(std::move(NextBlock));
```

Comment:
1. Here you're adding `NextBlock` unconditionally. If we have an instruction `brcond condition, target1, target2` that never falls through, we'll add next block too. Weird but not impossible.
2. Instead of checking `TargetPair.NextInstructionAddress != BasicBlock.nextBlock` twice, we could purge it after the fact.

Reply:
1. Ah, right. Thanks for spotting that.
2. I'm not the biggest fan of that approach, but it might be the right thing to do here. Considering it also solves the (1st) problem implicitly.

3. Also, now that I look into this again, I have a bit of FUD for the case where a function calls the next basic block (the case we ran into the other day)! Wouldn't we end up with duplicated targets then? (Leaving this here as a remainder to myself to make sure it's added at most once).

4. On that note, we also don't handle indirect calls gracefully. Imagine the disappointment of seeing the indirect call instruction as a link only for it to lead to the next instruction. (also just a reminder).

Reply:
1. OK.
2. OK.
3. If you think we should never return the same target twice, that's fine. Then return a set and state that we return a set in the docs. I don't think there's a benefit in returning the same location more than once here.
4. Eh, I don't see any way to represent in the UI "and more targets are available but we don't know about them". So I guess we'll just have to deal with that.

Reply:
3. Yeah, that's pretty much what I meant. The only duplication we can see is the fallthrough induced one, so there's not much reason to keep it around.

4. Ah, no, that wasn't what I was talking about. I mean the literal case where we have a `call rax` line and Ctrl+clicking it leads straight to the next instruction, which is is sad at best. What I meant is that in those cases we probably don't want any targets attached to it at all.

Reply:
I don't know, these really seem corner case that are taking too much time and discussion.

These are not important, do what makes the code simpler and simpler to describe without too many "unless..." and "ifs...".

Reply:
Done.